pgxn/neon/neon.c

```diff
@@ -522,29 +539,44 @@ RestoreRunningXactsFromClog(CheckPoint *checkpoint, TransactionId **xids, int *n
 			elog(LOG, "too many running xacts to restore from the CLOG; oldestXid=%u oldestActiveXid=%u nextXid %u",
 				 checkpoint->oldestXid, checkpoint->oldestActiveXid,
 				 XidFromFullTransactionId(checkpoint->nextXid));
-			goto fail;
+
+			overflow = true;
+			switch (running_xacts_overflow_policy)
+			{
+				case OP_WAIT:
```

Comment:
Am I right that the current behavior is 'wait' in this notation? What is the value we should set when we roll out this change? Is it 'ignore' or skip?

Reply:
Yes, current behaviour is "wait". In case of overflown snapshot, running xacts are not restored from CLOG and replica wait non-overflown running-xacts records from master.
If master in not active, then replica is stuck.

Yes, suggested value for replay is "ignore": extract as much running xacts from CLOG as possible and complete normal replica startup without waiting for master.

"skip" was added just for copmpleteness: it doesn't restore anything and complete startup. It is behaviour of Neoin before this PR with restoring XIDs from CLOG was merged.